ecosystem/node-checker/src/evaluators/direct/api/transaction_presence.rs

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{super::DirectEvaluatorInput, ApiEvaluatorError, API_CATEGORY};
+use crate::{
+    configuration::EvaluatorArgs,
+    evaluator::{EvaluationResult, Evaluator},
+    evaluators::EvaluatorType,
+};
+use anyhow::{anyhow, Result};
+use aptos_rest_client::{aptos_api_types::TransactionInfo, Client as AptosRestClient, Transaction};
+use aptos_sdk::crypto::HashValue;
+use clap::Parser;
+use poem_openapi::Object as PoemObject;
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+
+#[derive(Clone, Debug, Deserialize, Parser, PoemObject, Serialize)]
+pub struct TransactionPresenceEvaluatorArgs {
+    #[clap(long, default_value_t = 5)]
+    pub transaction_fetch_delay_secs: u64,
+}
+
+#[allow(dead_code)]
+#[derive(Debug)]
+pub struct TransactionPresenceEvaluator {
+    args: TransactionPresenceEvaluatorArgs,
+}
+
+impl TransactionPresenceEvaluator {
+    pub fn new(args: TransactionPresenceEvaluatorArgs) -> Self {
+        Self { args }
+    }
+
+    /// Get the transaction info of the latest transaction.
+    async fn get_transaction_info(
+        client: &AptosRestClient,
+    ) -> Result<TransactionInfo, ApiEvaluatorError> {
+        Self::unwrap_transaction_info(
+            client
+                .get_transactions(None, Some(1))
+                .await
+                .map_err(|e| {
+                    ApiEvaluatorError::EndpointError(
+                        "/transactions".to_string(),
+                        e.context(
+                            "The baseline API endpoint failed to return a transaction".to_string(),
+                        ),
+                    )
+                })?
+                .into_inner()
+                .into_iter()
+                .next()
+                .ok_or_else(|| {
+                    ApiEvaluatorError::EndpointError(
+                        "/transactions".to_string(),
+                        anyhow!("The baseline API returned"),
+                    )
+                })?,
+        )
+    }
+
+    /// Fetch a transaction by hash and return its transaction info.
+    async fn get_transaction_info_from_hash(
+        client: &AptosRestClient,
+        hash: HashValue,
+    ) -> Result<TransactionInfo, ApiEvaluatorError> {
+        Self::unwrap_transaction_info(
+            client
+                .get_transaction(hash)
+                .await
+                .map_err(|e| {
+                    ApiEvaluatorError::EndpointError(
+                        "/transactions".to_string(),
+                        e.context(
+                            format!("The baseline API endpoint failed to return the requested transaction: {}", hash),
```

Comment:
This could be called by the target client, right? So we shouldn't be mentioning `baseline` here?